docs/reference/indices/ignore-missing-component-templates.asciidoc

```diff
@@ -0,0 +1,102 @@
+[[ignore_missing_component_templates]]
+== Config ignore_missing_component_templates
+
+The configuration option `ignore_missing_component_templates` can be used when an index template references a component template that might not exist yet. Every time a data stream is create based on the index template, the existance of the component template will be checked. If it exists, it will taken as part of the mapping. If it does not exist, it is ignored.
+
+=== Usage example
+
+In the following, one component template and an index template are created. The index template references 2 component templates but only the `@package` one exists.
+
+
+Create of the component template `logs-foo@package`. This has to be created before the index template as it is not optional:
+
+// tag::ignore_missing_component_templates[]
+
+[source,console]
+----
+PUT _component_template/logs-foo@package
+{
+  "template": {
+    "mappings": {
+      "properties": {
+        "host.name": {
+          "type": "keyword"
+        }
+      }
+    }
+  }
+}
+----
+// TEST[continued]
+// end::ignore_missing_component_templates[]
+
+The index template is now created and and it references two component templates:
+
+[source]
+----
+  "composed_of": ["logs-foo@package", "logs-foo@custom"]
+----
+
+Before, only the `logs-foo@package` compontent template was created, meaning the `logs-foo@custom` is missing. Because of this the following entry was added to the config:
+
+[source]
+----
+  "ignore_missing_component_templates": ["logs-foo@custom"],
+----
+
+During creation of the template, it will not validate that `logs-foo@custom` exists:
+
+
+[source,console]
+----
+PUT index_template/logs-foo
+{
+  "index_patterns": ["logs-foo-*"],
+  "data_stream": { },
+  "composed_of": ["logs-foo@package", "logs-foo@custom"],
+  "ignore_missing_component_templates": ["logs-foo@custom"],
+  "priority": 500
+}
+----
+// TEST[continued]
+// end::ignore_missing_component_templates[]
+
+The index template `logs-foo` was successfully created. A data stream can be created based on this template:
+
+[source,console]
+----
+PUT _data_stream/logs-foo-bar
+----
+// TEST[continued]
+// end::ignore_missing_component_templates[]
+
+Looking at the mappings of the data stream, it will contain the `host.name` field.
+
+At a later stage, the missing component template might be added:
+
+[source,console]
+----
+PUT _component_template/logs-foo@custom
+{
+  "template": {
+    "mappings": {
+      "properties": {
+        "host.ip": {
+          "type": "ip"
+        }
+      }
+    }
+  }
+}
+----
+// TEST[continued]
+// end::ignore_missing_component_templates[]
+
+This will not have an immediate effect on the data stream. The mapping `host.ip` will only show up in the data stream mappings when the data stream is rolled over automatically next time or a manual rollover is triggered:
+
+[source,console]
+----
+POST logs-foo-bar/_rollover
+----
+// TEST[continued]
```

Comment:
```suggestion
// TEST[continued]
// TEST[teardown:data_stream_cleanup]
```

Reply:
At the end of a page with code snippets that create index patterns, you need to clean up those index patterns. The line `// TEST[teardown:data_stream_cleanup]` after the last snippet on the page will do that. It refers to this teardown: https://github.com/elastic/elasticsearch/blob/main/docs/build.gradle#L210

Reply:
Added